userland/libtock/adc.c

```diff
@@ -18,12 +25,33 @@ static void adc_cb(__attribute__ ((unused)) int callback_type,
   struct adc_data* data = (struct adc_data*) ud;
   data->reading = reading;
   data->fired = true;
+
+  // In continuous mode
+  if (cont_cb)
```

Comment:
I really like the idea that you had here of masking the messing `adc_interval_cb` type and allowing the user to pass in their own simpler callback function. I'm definitely going to pull that idea into #371